twilio_integration/overrides/notification.py

```diff
@@ -1,38 +1,127 @@
 import frappe
 from frappe import _
-from frappe.email.doctype.notification.notification import Notification, get_context, json
 from twilio_integration.twilio_integration.doctype.whatsapp_message.whatsapp_message import WhatsAppMessage
+from frappe.core.doctype.notification_count.notification_count import set_notification_last_scheduled
+from frappe.email.doctype.notification.notification import (
+    Notification,
+    get_context,
+    json,
+    get_reference_doctype,
+    get_reference_name,
+)
 
 class SendNotification(Notification):
 	def validate(self):
 		self.validate_twilio_settings()
 
 	def validate_twilio_settings(self):
-		if self.enabled and self.channel == "WhatsApp" \
-			and not frappe.db.get_single_value("Twilio Settings", "enabled"):
-			frappe.throw(_("Please enable Twilio settings to send WhatsApp messages"))
+		if self.enabled and self.channel == "WhatsApp":
+			twilio_settings = frappe.get_doc("Twilio Settings")
+			if not twilio_settings.enabled:
+				frappe.throw(_("Twilio Settings must be enabled to send WhatsApp notifications."))
+			if not twilio_settings.whatsapp_no:
+				frappe.throw(_("Twilio WhatsApp Number is required in Twilio Settings."))
+
+	def send(self, doc, context=None):
+
+		if not context:
+			context = get_context(doc)
+
+		context.update({"doc": doc, "alert": self, "comments": None})
 
-	def send(self, doc):
-		context = get_context(doc)
-		context = {"doc": doc, "alert": self, "comments": None}
 		if doc.get("_comments"):
 			context["comments"] = json.loads(doc.get("_comments"))
 
 		if self.is_standard:
 			self.load_standard_properties(context)
 
 		try:
-			if self.channel == 'WhatsApp':
+			if self.channel == 'WhatsApp' and self.enabled:
 				self.send_whatsapp_msg(doc, context)
-		except:
-			frappe.log_error(title='Failed to send notification', message=frappe.get_traceback())
+		except Exception as e:
+			frappe.log_error(
+				message=frappe.get_traceback(),
+				title=_("Failed to send WhatsApp Notification: {0} for {1} {2}").format(self.name, doc.doctype, doc.name)
```

Comment:
Instead of mentioning doctype and docname in title, pass reference doctype and name